core/api.js

```diff
@@ -0,0 +1,54 @@
+import * as composer from "./composer";
+
+function flattenInputs(response) {
+  // duplicate inputs exist when more than one build is available
+  // flatten duplicate inputs to a single item
+  const previousInputs = {};
+  const flattened = response.filter((item) => {
+    const build = {
+      version: item.builds[0].source.version,
+      release: item.builds[0].release,
+    };
+    if (previousInputs.hasOwnProperty(item.name)) {
+      // update the previousInput object with this item"s version/release
+      // to make the default version/release the latest
+      previousInputs[item.name] = Object.assign(previousInputs[item.name], build);
+      // and remove this item from the list
+      return false;
+    }
+    delete item.builds;
+    item = Object.assign(item, build);
+
+    previousInputs[item.name] = item;
+    return true;
+  });
+  return flattened;
+}
+
+async function getPackages(filter, selectedInputPage, pageSize) {
+  try {
+    const wildcardsUsed = filter.includes("*");
+    const regex = / +|, +/g;
+    let filterValue = filter.replace(regex, ",");
+    const regexStrip = /(^,+)|(,+$)/g;
+    filterValue = filterValue.replace(regexStrip, "");
+    filterValue = wildcardsUsed ? filterValue : `*${filterValue}*`.replace(/,/g, "*,*");
```

Comment:
I don't know what the purpose behind every line is because as you pointed out, it has just been copy/pasted it so that the the search results would be the same as before. But overall the purpose is to transform the search query into a form that will actually work with the Weldr API which expects you to use things like leading and trailing wildcards. So for instance, if you want to search for "vim", you actually need to query the Weldr API "*vim*" if you want it to behave as you would expect show packages such as neovim.

Reply:
- `wildcardsUsed` is true if an `*` appears in the filter
- `regex` matches a non-empty sequence of spaces, or a comma followed by a non-empty sequence of spaces
- `filterValue` replaces each match of `regex` with a comma - in other words all spaces are replaced by commas, without inserting repeated commas
- `regexStrip` matches any non-empty sequence of commas at the beginning or end of a string
- `filterValue` is update to replace leading or trailing strings of commas with the empty string
- finally, as Lucas said, each match is replaced with wildcard versions, except if wildcards are already used.

I wonder if a future task could be to make weldr search better, if we push the search query into `dnf-json` we would get more powerful searching (making this logic redundant) and potentially much faster queries.